pkg/networkmanager/network_manager.go

```diff
@@ -293,19 +284,27 @@ func (am *NetworkManager) monitorContainer(ctx context.Context, container *conta
 				watchedContainer.UpdateDataTicker.Reset(am.cfg.UpdateDataPeriod)
 			}
 
-			am.handleNetworkEvents(ctx, container, watchedContainer)
+			watchedContainer.SetStatus(utils.WatchedContainerStatusReady)
+			am.saveNetworkEvents(ctx, container, watchedContainer)
 		case err := <-watchedContainer.SyncChannel:
 			switch {
 			case errors.Is(err, utils.ContainerHasTerminatedError):
-				am.handleNetworkEvents(ctx, container, watchedContainer)
+				// if exit code is 0 we set the status to completed
+				// TODO: Should we split ContainerHasTerminatedError to indicate if we reached the maxSniffingTime?
+				if watchedContainer.GetTerminationExitCode(am.k8sObjectCache, container.K8s.Namespace, container.K8s.PodName, container.K8s.ContainerName) == 0 {
+					watchedContainer.SetStatus(utils.WatchedContainerStatusCompleted)
+				}
+				am.saveNetworkEvents(ctx, container, watchedContainer)
 				return nil
 			}
+
+			// TODO: @Amir should we handle too large?
```

Comment:
@dwertent should we support the "too large" status also for NN?

Reply:
We should potentially, but we didn't define it (we did not encounter large NP)